src/devices/steelmate.c

```diff
@@ -52,19 +55,22 @@ Bytes 2 to 9 are inverted Manchester with swapped MSB/LSB:
 
 static int steelmate_callback(r_device *decoder, bitbuffer_t *bitbuffer)
 {
+    uint8_t const preamble_pattern[] = {0x00, 0x00, 0x7f}; // inverted, raw value is 0x5a
     //Loop through each row of data
-    for (int row = 0; row < bitbuffer->num_rows; row++)
-    {
+    for (int row = 0; row < bitbuffer->num_rows; row++) {
         //Payload is inverted Manchester encoded, and reversed MSB/LSB order
-        uint8_t *b = bitbuffer->bb[row];
+        uint8_t *b       = bitbuffer->bb[row];
+        unsigned row_len = bitbuffer->bits_per_row[row];
 
-        //Length must be 72 bits to be considered a valid packet
-        if (bitbuffer->bits_per_row[row] != 72)
+        //Length must be 72, 73, 208 or 209 bits to be considered a valid packet
+        if (row_len != 72 && row_len != 73 && row_len != 209 && row_len != 208)
             continue; // DECODE_ABORT_LENGTH
 
         //Valid preamble? (Note, the data is still wrong order at this point. Correct pre-amble: 0x00 0x00 0x01)
-        if (b[0] != 0x00 || b[1] != 0x00 || b[2] != 0x7f)
+        unsigned bitpos = bitbuffer_search(bitbuffer, row, 0, preamble_pattern, 24);
+        if (bitpos >= row_len - 64)
             continue; // DECODE_ABORT_EARLY
+        b = &b[bitpos / 8];
```

Comment:
We should either `bitbuffer_extract_bytes()` or make sure `bitpos % 8 == 0`

Reply:
Updated to use better bit wrangling.